wagtail/images/models.py

```diff
@@ -1384,16 +1384,35 @@ def get_upload_to(self, filename):
     def check(cls, **kwargs):
         errors = super().check(**kwargs)
         if not cls._meta.abstract:
-            if not any(
+            in_unique_together = any(
                 set(constraint) == {"image", "filter_spec", "focal_point_key"}
                 for constraint in cls._meta.unique_together
-            ):
+            )
+
+            # We only include fields to check if they are a UniqueConstraint, and
+            # they have no conditions on them.
+            unique_constraint_fields = [
+                constraint.fields
+                for constraint in cls._meta.constraints
+                if (
+                    isinstance(constraint, UniqueConstraint)
+                    and constraint.condition is None
+                )
+            ]
+            in_unique_constraint = any(
+                set(constraint) == {"image", "filter_spec", "focal_point_key"}
+                for constraint in unique_constraint_fields
+            )
+
+            # We check for some kind of unique constraint, whether in unique together or
+            # as part of the constraints.
+            if not in_unique_together and not in_unique_constraint:
```

Comment:
I considered adding another check to warn when both `unique_together` and `UniqueConstraint` is used (like we did for `TranslatableMixin`). However, I think this is unlikely to happen because unlike `TranslatableMixin`, we don't put a default `unique_together` in `AbstractRendition`, so the option is completely up to the developers.